cloudinit/config/cc_update_hostname.py

```diff
@@ -74,6 +75,14 @@
         prefer_fqdn_over_hostname: false
         """
         ),
+        dedent(
+            """\
+        # On a machine without an ``/etc/hostname`` file, don't create it
+        # In most clouds, this will results in a DHCP configured hostname
+        # provided by the cloud
```

Comment:
I didn't make these changes ("this will result in a hostname provided by the cloud" -> "this will result in a hostname being provided by the cloud") because the implication of the second on is, to me, that the key we provide has an effect on the cloud DHCP service, which it does not.  The original statement is grammatically correct in my opinion (provided by the cloud modifies hostname) and is more appropriate to the situation.  Let me know if you feel differently/would like me to reword the comment!

Reply:
I'm not sure I agree, but if that's how you interpreted it then chances are other people might also, so happy to leave as is. Thanks for clarifying! Feel free to mark these as resolved.